include/library/spdm_device_secret_lib.h

```diff
@@ -226,3 +226,35 @@ bool libspdm_psk_master_secret_hkdf_expand(
  **/
 bool libspdm_write_certificate_to_nvm(uint8_t slot_id, const void * cert_chain,
                                       size_t cert_chain_size);
+
+/**
+ * save the CSR
+ *
+ * @param[out] csr_len               CSR len for DER format
+ * @param[in]  csr_pointer           csr_pointer is address to store CSR.
+ * @param[in]  file_name             file name
+ *
+ * @retval true                      successfully.
+ * @retval false                     unsuccessfully.
+ **/
+bool libspdm_write_csr_to_nvm(const void * csr_pointer, size_t csr_len, char *file_name);
```

Comment:
why we need file_name???

Reply:
I have changed the API to  test_crypt internal.